adapters/smartx/smartx.go

```diff
@@ -0,0 +1,90 @@
+package smartx
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpointURL string
+}
+
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpointURL: config.Endpoint,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) (requestsToBidder []*adapters.RequestData, errs []error) {
+	openRTBRequestJSON, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("marshal bidRequest: %w", err))
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if openRTBRequest.Device != nil {
+		if openRTBRequest.Device.UA != "" {
+			headers.Set("User-Agent", openRTBRequest.Device.UA)
+		}
+
+		if openRTBRequest.Device.IP != "" {
+			headers.Set("Forwarded", "for="+openRTBRequest.Device.IP)
+			headers.Set("X-Forwarded-For", openRTBRequest.Device.IP)
+		}
+	}
+
+	return append(requestsToBidder, &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpointURL,
+		Body:    openRTBRequestJSON,
+		Headers: headers,
+	}), nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
```

Comment:
you might wanna reverse this. You should check for no content first before you check `CheckResponseStatusCodeForErrors` because `CheckResponseStatusCodeForErrors` function will return an error even if response from bidder is `204` which is StatusNoContent. So - 
```
if adapters.IsResponseStatusCodeNoContent(responseData) {
		return nil, nil
}
if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
		return nil, []error{err}
}